staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1450,9 +1485,6 @@ func mergeSlice(original, patch []interface{}, schema LookupPatchMeta, mergeKey
 	kind := t.Kind()
 	// If the elements are not maps, merge the slices of scalars.
 	if kind != reflect.Map {
-		if mergeOptions.MergeParallelList && isDeleteList {
```

Comment:
This option is gone somewhere, was it supposed to?

Reply:
Yeah, I don't think it actually applies when you are using the $deleteFromPrimitiveList directive, so I excluded it from my refactor.

But on second thought, my intention is not to change any behavior, other than to fix the original bug, so I updated the PR to keep performing this check.